### PR TITLE
[test] Remove old workarounds for Jacobi test

### PR DESCRIPTION
**Pull request by jgphpc**
https://jira.cscs.ch/browse/PROGENV-12

### CODE REVIEW

File: cscs-checks/tools/profiling_and_debugging/notool.py [hunk 1]
```diff
@@ -47,94 +47,64 @@ def restrict_f90_concurrency(self):
...
-            self.variables = {
-                'PKG_CONFIG_PATH': f'$PKG_CONFIG_PATH:{mpich_pkg_config_path}'
-            }
-            self.cppflags = ('`pkg-config --cflags mpich` '
-                             '`pkg-config --libs mpich`')
-        elif sysname in ['pilatus', 'eiger']:
-            self.skip_if(self.current_environ.name == 'PrgEnv-nvidia', '')
+        if sysname in ['pilatus', 'eiger', 'dom', 'daint']:
+            # FIXME: Fatal error in PMPI_Init_thread
+            self.skip_if(envname == 'PrgEnv-nvidia', '')
```
Comment (vkarak) [inline cscs-checks/tools/profiling_and_debugging/notool.py:62]: I'd suggest to add a message why you are skipping it.

File: cscs-checks/tools/profiling_and_debugging/notool.py [hunk 2]
```diff
@@ -47,94 +47,64 @@ def restrict_f90_concurrency(self):
...
 
     @run_before('compile')
     def set_flags(self):
         # FIXME: workaround for C4KCUST-308
-        self.modules += ['cray-mpich']
+        # self.modules += ['cray-mpich']
```
Comment (vkarak) [inline cscs-checks/tools/profiling_and_debugging/notool.py:67]: Remove these two completely, if they are not actually needed anymore.

File: cscs-checks/tools/profiling_and_debugging/notool.py [hunk 3]
```diff
@@ -47,94 +47,64 @@ def restrict_f90_concurrency(self):
...
+            'PrgEnv-nvidia': ['-O2', '-g', '-mp']
         }
         envname = self.current_environ.name
         # if generic, falls back to -g:
         prgenv_flags = self.prgenv_flags.get(envname, ['-g'])
         self.build_system.cflags = prgenv_flags
         self.build_system.cxxflags = prgenv_flags
         self.build_system.fflags = prgenv_flags
-        self.build_system.ldflags = ['-lm']
+        # self.build_system.ldflags = ['-lm']
```
Comment (vkarak) [inline cscs-checks/tools/profiling_and_debugging/notool.py:84]: Remove?

**Review by vkarak — CHANGES_REQUESTED** [review]
Can you update the PR title and the description. It doesn't seem to correspond to what this PR exactly does. Other than that, I have some minor comments only.

File: cscs-checks/tools/profiling_and_debugging/notool.py (hunk 1 above)
Comment (jgphpc) [reply cscs-checks/tools/profiling_and_debugging/notool.py:62]: I get The same issue when running `helloworld.py` on Alps (notice that `PrgEnv-nvidia` is not `valid_prog_environs`)
On alps, you'll get this issue with `PrgEnv-nvidia`:
```
MPICH ERROR [Rank 0] [job id ] [Sat Dec  4 13:09:40 2021] [nid001433] - 
Abort(1112591) (rank 0 in comm 0): Fatal error in PMPI_Init: 
Other MPI error, error stack:
MPIR_Init_thread(144):
MPID_Init(430).......:
MPIR_pmi_init(83)....: PMI2_Job_GetId returned 14
```

File: cscs-checks/tools/profiling_and_debugging/notool.py (hunk 3 above)
Comment (jgphpc) [reply cscs-checks/tools/profiling_and_debugging/notool.py]: https://github.com/eth-cscs/reframe/pull/2298/commits/94516b25e5c7b6d8dbf933c18df2680db76c0608

File: cscs-checks/tools/profiling_and_debugging/notool.py (hunk 2 above)
Comment (jgphpc) [reply cscs-checks/tools/profiling_and_debugging/notool.py]: https://github.com/eth-cscs/reframe/pull/2298/commits/94516b25e5c7b6d8dbf933c18df2680db76c0608

File: cscs-checks/tools/profiling_and_debugging/notool.py (hunk 1 above)
Comment (jgphpc) [reply cscs-checks/tools/profiling_and_debugging/notool.py:62]: https://github.com/eth-cscs/reframe/pull/2298/commits/94516b25e5c7b6d8dbf933c18df2680db76c0608

Comment (vkarak) [reply cscs-checks/tools/profiling_and_debugging/notool.py:62]: I meant add a message in the `skip_if` statement, something like "skipping buggy environment".

**Review by vkarak — APPROVED** [review]

**Comment by victorusu** [comment]
@jgphpc, I just do not understand why you are removing the files from the other tests, not the `notool.py` one...

**Comment by vkarak** [comment]
@victorusu I guess they were forgotten, because all of @jgphpc's  tests have been moved to his repo in the past. @jgphpc Can you confirm?

**Review by victorusu — APPROVED** [review]
`lgtm`

**Comment by jgphpc** [comment]
> Can you confirm?
yes, it's written in the file.

**Pull request merged**